CredentialsManager/Models/Menu.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CredentialsManager.Models
+{
+  public class Menu
+  {
+    public string Header { get; set; }
+    public IList<MenuItem> MenuItems { get; set; }
+
+    public MenuItem ExitItem { get; set; }
+
+    public Menu()
+    {
+    }
+
+    public Menu(string header, IList<MenuItem> menuItems, MenuItem exitItem)
+    {
+      Header = header ?? throw new ArgumentNullException(nameof(header));
```

Comment:
Personally I only do this kind of validation on input that comes from external sources (users, files, etc.)  Even if you later add the ability to (for example) create a menu from a file, you should be doing the validation when you read the menu file and not here.

Reply:
I will definitely give more thought next time I do validation inside a constructor.
Thank you.